ls1intum/Artemis
# `Programming exercises`: Fix an issue when pushing to an instructor's assignment repository

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
https://github.com/ls1intum/Artemis/issues/6391

### Description
Remove unnecessary code for authorizing pushes from the online editor and adapt the `checkRepositoryAccess()` method to allow editors and higher for a course to push to any repository of that course even after the due date.

### Steps for Testing
Prerequisites:
- 1 Instructor or Editor
- 1 Student
- 1 Programming Exercise with no due date set

1. Log in to Artemis as a student.
2. Navigate to the online editor of the exercise and make sure you can push.
3. Log in as the instructor, go Course Management, find the exercise, and click on `Edit in editor`
4. Click on `Create Assignment Repository` and make sure you can push from there.
5. Edit the exercise and set the due date to be in the past.
6. Log in as the student, and make sure the repository is locked.
7. Log in as the instructor, and make sure that you can still push to your assignment repository that you can access via `Edit in editor`. Note: There is another way to access an online editor for your assignment repository (the same way a student would access their assignment repository via the course overview). This editor should display "Repository is locked".

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line | Comment |
|------------|-------:|-----:|---------:|
| RepositoryAccessService.java | 85% | 100% | all changes covered |
| RepositoryProgrammingExerciseParticipationResource.java | 72% | 81% | mostly removed code + some small changes that are fully covered |
